Update operator docs (#457)
wasmCloud/wasmCloud.com: docs/deployment/k8s/index.mdx

On **this** page, you can find documentation for operators wishing to deploy wasmCloud with Kubernetes, including details on prerequisites such as NATS and how to deploy the operator.

:::info[kubectl support]

> When running wasmCloud on Kubernetes, you can use `kubectl` to deploy [wasmCloud manifests](/docs/ecosystem/wadm/model), check wasmCloud logs, and more. The example below demonstrates standard usage of `kubectl` with wasmCloud.

Reviewer: I'm wondering if we've settled on the terminology here, I know we've called them "wadm applications" and "wasmcloud applications" interchangeably (or maybe that's just me?) in the past, but I don't know if we've got a preferred term to use.

Reviewer: by here I'm referencing `wasmCloud manifests` specifically

Author: My 2 cents is that "wasmCloud application manifests" is preferable. If you're new to the platform (running through the quickstart, say) understanding exactly what wadm is, what it stands for, etc probably shouldn't be a conceptual prerequisite for using a declarative manifest to deploy a thing on wasmCloud.

:::

## Getting started with the wasmCloud operator

> The wasmCloud operator can be found on GitHub at [https://github.com/wasmCloud/wasmcloud-operator](https://github.com/wasmCloud/wasmcloud-operator). It uses custom resource definitions (CRDs) to define wasmCloud hosts as Kubernetes resources.

Reviewer: It may be redundant given the `:::info:::` block above, but it might be worth enumerating the 3 things the Operator does for you today:

1. Enables deployment of wasmCloud Hosts into the cluster via Custom Resource Definition (`WasmCloudHostConfig`)
2. Enables interoperability with wadm for deploying wasmcloud applications defined in the [OAM spec](https://oam.dev/) format. 
3. Enables automated creation of Kubernetes Services pointing to wasmcloud applications deployed with a HTTP Server capability (in the future this will become configurable, today it's automatic).

In order to follow the instructions on this page, you will need a Kubernetes cluster and appropriate access credentials as well as a [Helm](https://helm.sh/) installation. It will also be useful to have [wasmCloud Shell (`wash`)](/docs/installation) installed locally. 

### Deploying the operator

A wasmCloud cluster requires a few things to run:

- A **[NATS](/docs/ecosystem/nats/) cluster** (with Jetstream enabled)

> - **[wasmCloud Application Deployment Manager (wadm)](/docs/ecosystem/wadm/)** connected to the NATS cluster in order to support applications

Reviewer: ```suggestion
- **[wasmCloud Application Deployment Manager (wadm)](/docs/ecosystem/wadm/)** connected to the NATS cluster in order to support deploying wasmCloud applications
```

    enabled: true
    fileStore:
      pvc:

>         size: 5Gi

Reviewer: I'm not specifically commenting on the size change, I think this is great, but we might want to include enabling the websockets listener in this config block for future `wash ui` or `satellite-ui` examples, which would effectively just mean adding this either right after `leafnodes`:
```yaml
config:
... <existing-config-goes-here>
  websocket:
    enabled: true
    port: 4223
```

helm upgrade --install -f values.yaml nats-cluster nats/nats
```

#### Deploying wadm

You can deploy wadm to your Kubernetes cluster with a Helm chart hosted by wasmCloud. Use this `values.yaml` file to define a minimal deployment that connects with the NATS server deployed above: 

```yaml
wadm:
  config:
    nats:
      server: "nats-cluster.default.svc.cluster.local:4222"
```

Run `helm install`:

```shell

> helm install wadm -f values.yaml --version 0.2.2 oci://ghcr.io/wasmcloud/charts/wadm

Reviewer: In the interest that we improve the wadm chart in the future, I might recommend we drop `--version 0.2.2` to have the end-user pick up the latest version, wdyt?
```suggestion
helm install wadm -f values.yaml oci://ghcr.io/wasmcloud/charts/wadm
```

```

#### Start the operator

To download and deploy the operator:

```shell

> helm install wasmcloud-operator --version 0.1.3 oci://ghcr.io/wasmcloud/charts/wasmcloud-operator

Reviewer: Same comment as above for `wadm`, as I'm sure we'll keep improving the wasmcloud-operator chart
```suggestion
helm install wasmcloud-operator oci://ghcr.io/wasmcloud/charts/wasmcloud-operator
```

    # `wash keys gen cluster`
    - <public-key>
  # The lattice to connect the hosts to
  lattice: 83a5b52e-17cf-4080-bac8-f844099f142e
  # Additional labels to apply to the host other than the defaults set in the operator
  hostLabels:
    some-label: value
  # The address to connect to nats 
  natsAddress: nats://nats-cluster.default.svc.cluster.local
  # Which wasmCloud version to use
  version: 1.0.2
  # The name of a secret in the same namespace that provides the required secrets.
  secretName: my-wasmcloud-cluster
  # Enable the following to run the wasmCloud hosts as a DaemonSet
  #daemonset: true
  # The name of the image pull secret to use with wasmCloud hosts so that they
  # can authenticate to a private registry to pull components.
  # registryCredentialsSecret: my-registry-secret
```
```shell
kubectl apply -f config.yaml
```

Next we will create a manifest called `secret.yaml` for our secret, adding the seed generated above in the `WASMCLOUD_CLUSTER_SEED` field:

```yaml
apiVersion: v1
kind: Secret
metadata:
  name: my-wasmcloud-cluster
stringData:
  # You can generate this with wash:
  # `wash keys gen cluster`
  WASMCLOUD_CLUSTER_SEED: <seed>

> #data:

Reviewer: I would probably just remove the `#data:` and the lines following it entirely, I don't know it adds anything in to the example

  # Only required if using a NATS creds file
  # nats.creds: <creds file>
```
```shell
kubectl apply -f secret.yaml
```

You can confirm that your wasmCloud hosts have launched by checking logs for a given host with:

```shell
kubectl logs my-wasmcloud-cluster-xyz -c wasmcloud-host
```

> (You can find your wasmCloud host pod's name with `kubectl get pods`.)

Reviewer: Rather than having to look up the specific pods name, you could also use a label selector like so:
```
kubectl logs -f -l app.kubernetes.io/instance=my-wasmcloud-cluster -c wasmcloud-host
```
This has the downside of getting logs from both pods, but they should look roughly the same assuming things have gone right!

Author: Much more elegant, thank you!

### Deploying a workload

> When you `kubectl apply` a wasmCloud manifest, the cluster will automatically provision the component workload with wasmCloud. The example below uses a "hello world" manifest saved to a file called `wadm.yaml`:

Reviewer: ```suggestion
When you `kubectl apply` a wasmCloud application, the cluster will automatically provision the component workload with wasmCloud. The example below uses a "hello world" manifest saved to a file called `wadm.yaml`:

```

        # Your manifest deploys from a local .wasm file, but you can also use OCR registries as below
        image: wasmcloud.azurecr.io/http-hello-world:0.1.0
      traits:
        # One replica of this component will run
        - type: daemonscaler
          properties:
            replicas: 1
    # The httpserver capability provider, started from the official wasmCloud OCI artifact
    - name: httpserver
      type: capability
      properties:
        image: ghcr.io/wasmcloud/http-server:0.20.0
      traits:
        # Link the HTTP server and set it to listen on the local machine's port 8080
        - type: link
          properties:
            target: http-component
            namespace: wasi
            package: http
            interfaces: [incoming-handler]
            source_config:
              - name: default-http
                properties:
                  ADDRESS: 127.0.0.1:8080

```

Run `kubectl apply`:

```shell
kubectl apply -f wadm.yaml
application/hello-world created
```

Check the deployment status:

```shell
kubectl get application
APPLICATION   DEPLOYED VERSION   LATEST VERSION   STATUS
hello-world   v0.0.1             v0.0.1           Deployed
```

> Port-forward the NATS service on your Kubernetes cluster to your local machine:

Reviewer: ```suggestion
Port-forward from your local machine to the NATS service running in your Kubernetes cluster:
```
